From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LinkBoi00 <linkdevel@protonmail.com>
Date: Sun, 14 Aug 2022 00:23:54 +0300
Subject: [PATCH] light: Adapt brightness scaling behaviour to match MIUI

MIUI's lowest brightness value is '19' and using lower values
can cause the backlight to completely power off inadvertedly

Therefore, let's just account for this behaviour in the light AIDL
by adding '19' instead of '1' to the final brightness value

[fukiame: cap at 25 instead]

Change-Id: I0e31bda9a0cc505f1246209810eae624dfe5d068

Signed-off-by: LinkBoi00 <linkdevel@protonmail.com>
Signed-off-by: Ookiineko <chiisaineko@protonmail.com>
Signed-off-by: fukiame <fukiame@proton.me>

diff --git a/light/Utils.cpp b/light/Utils.cpp
index 95626a7..9e2493a 100644
--- a/light/Utils.cpp
+++ b/light/Utils.cpp
@@ -45,7 +45,11 @@ uint8_t rgb::toBrightness() {
 }
 
 uint32_t scaleBrightness(uint8_t brightness, uint32_t maxBrightness) {
-    return brightness * maxBrightness / 0xFF;
+    if (brightness == 0) {
+        return 0;
+    }
+
+    return (brightness - 1) * (maxBrightness - 25) / (0xFF - 1) + 25;
 }
 
 }  // namespace light
